src/stream/src/executor/source.rs

```diff
@@ -61,6 +61,8 @@ pub struct SourceExecutor<S: StateStore> {
     split_state_store: SourceStateHandler<S>,
     // store latest split to offset mapping
     state_cache: Option<Vec<ConnectorState>>,
+    // whether there is any data flowing between two barriers
+    idle_between_barriers: bool,
```

Comment:
it's unnecessary to add a field here. we can make `state_cache = None` after flushing it to the state store. If `state_cache` is None, the flush can be skipped.

Reply:
Good point. Done.